src/common/transformations/src/transformations/common_optimizations/shared_ops_optimization.cpp

```diff
@@ -163,6 +170,45 @@ bool converts_are_equal(const Node* lhs, const Node* rhs) {
            inputs_from_same_source_or_equal_constants(lhs, rhs);
 }
 
+bool divide_are_equal(const Node* lhs, const Node* rhs) {
+    const auto l_div = as_type<const v1::Divide>(lhs);
+    const auto r_div = as_type<const v1::Divide>(rhs);
+    if (!l_div || !r_div)
+        return false;
+    return l_div->get_autob() == r_div->get_autob() && l_div->is_pythondiv() == r_div->is_pythondiv() &&
+           inputs_from_same_source_or_equal_constants(lhs, rhs);
+}
+
+bool eltwise_arithmethic_are_equal(const Node* lhs, const Node* rhs) {
+    const auto l_eltwise = as_type<const op::util::BinaryElementwiseArithmetic>(lhs);
+    const auto r_eltwise = as_type<const op::util::BinaryElementwiseArithmetic>(rhs);
+    if (!l_eltwise || !r_eltwise)
+        return false;
+    return l_eltwise->get_autob() == r_eltwise->get_autob() && inputs_from_same_source_or_equal_constants(lhs, rhs);
+}
+
+bool pads_are_equal(const Node* lhs, const Node* rhs) {
+    const auto l_pad = as_type<const op::util::PadBase>(lhs);
+    const auto r_pad = as_type<const op::util::PadBase>(rhs);
+    if (!l_pad || !r_pad)
+        return false;
+    return l_pad->get_pad_mode() == r_pad->get_pad_mode() && inputs_from_same_source_or_equal_constants(lhs, rhs);
+}
+
+bool max_pools_8_are_equal(const Node* lhs, const Node* rhs) {
+    const auto l_pool = as_type<const v8::MaxPool>(lhs);
+    const auto r_pool = as_type<const v8::MaxPool>(rhs);
+    if (!l_pool || !r_pool)
+        return false;
+    return l_pool->get_autob() == r_pool->get_autob() && l_pool->get_axis() == r_pool->get_axis() &&
```

Comment:
I don't see auto broadcasting for Max Pool
```suggestion
    return l_pool->get_axis() == r_pool->get_axis() &&
```